File: frontend/src/domain/fieldOfLaw.ts
export type Norm = {
  abbreviation: string
  singleNormDescription: string
}

export type FieldOfLaw = {
  identifier: string
  text: string
  linkedFields?: string[]
  norms: Norm[]
  children: FieldOfLaw[]
  parent?: FieldOfLaw
  hasChildren: boolean
}

export function buildRoot(): FieldOfLaw {
  return {
    identifier: "root",
    text: "Alle Sachgebiete",
    children: [],
    norms: [],
    hasChildren: true,
  }
}

export function createNode(identifier: string): FieldOfLaw {
  return {
    identifier: identifier,
    text: "",
    children: [],
    norms: [],
    hasChildren: false,
  }
}

File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/lookuptable/fieldoflaw/FieldOfLaw.java
package de.bund.digitalservice.ris.caselaw.domain.lookuptable.fieldoflaw;

import java.util.List;
import java.util.UUID;
import javax.annotation.Nullable;
import lombok.Builder;

@Builder(toBuilder = true)
public record FieldOfLaw(
    UUID id,
    boolean hasChildren,
    String identifier,
    String text,
    String notation,
    List<String> linkedFields,
    List<Norm> norms,
    List<FieldOfLaw> children,
    @Nullable FieldOfLaw parent) {}
